test_cli.py

```diff
@@ -112,9 +114,43 @@ def sanitize(string, max_chars=256):
     return string
 
 
+class FakeStdout(io.StringIO):
+    """Fake class to mimic stdout. We can't just use io.StringIO because we need
+    to fake the ability to write binary files to sys.stdout.buffer (thus this
+    class has a "buffer" attribute that behaves the same way).
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.buffer = io.BytesIO()
+
+    def getvalue(self):
+        """If self.buffer has a non-unicode value, return that value.
+        Otherwise, decode the self.buffer value and append it
+        to self.getvalue().
+        """
+        try:
+            buffer_value = self.buffer.getvalue().decode()
+        except UnicodeDecodeError:
+            return self.buffer.getvalue()
+        return super().getvalue() + buffer_value
```

Comment:
I'm confused...this function can either return `str` or `bytes`?  I don't understand how both can be fine...

Reply:
This is because this function is mimicking the behavior of `sys.stdout`. `sys.stdout` can be read as either a string or bytes.

When a string is written to `sys.stdout`, it returns a string when doing `getvalue()`.

When bytes are written to `sys.stdout` (by writing to `sys.stdout.buffer`), it returns bytes when doing `getvalue()`. The reason I had to account for this case is that there are tests in which a binary file is uploaded, then it is printed out (by writing to `sys.stdout.buffer`), and then the test reads what's printed out and makes sure it matches the original file.

Reply:
Can you add this to the comment?